src/Modules/OrchardCore.Commerce/Events/InventoryCheckoutEvents.cs

```diff
@@ -39,10 +48,28 @@ public Task<bool> ViewModelCreatedAsync(IList<ShoppingCartLineViewModel> lines)
 
             if (cannotCheckout)
             {
-                return Task.FromResult(cannotCheckout);
+                if (checkoutViewModel != null)
+                {
+                    checkoutViewModel.IsInvalid = true;
+                }
+                else
+                {
+                    shoppingCartViewModel.InvalidReasons.Add(H["{0} is out of stock.", inventoryIdentifier]);
+                }
+
+                return Task.CompletedTask;
             }
         }
 
-        return Task.FromResult(cannotCheckout);
+        if (checkoutViewModel != null)
```

Comment:
Why is this necessary? I think if in the future we'll have other events that can invalidate the cart, then we shouldn't erase that here.